pkg/kubelet/kubelet_test.go

```diff
@@ -381,6 +385,102 @@ func newTestPods(count int) []*v1.Pod {
 	return pods
 }
 
+// For the watchdog test below we need something which blocks the sync loop for some time..
+// Use a SyncHandler which just sleeps when handling a pod addition
+type blockingAdditionHandler struct{}
+
+var blockDuration = 5 * time.Second
+
+func (f blockingAdditionHandler) HandlePodAdditions(pods []*v1.Pod) {
+	time.Sleep(blockDuration)
+}
+func (f blockingAdditionHandler) HandlePodUpdates(pods []*v1.Pod)   {}
+func (f blockingAdditionHandler) HandlePodRemoves(pods []*v1.Pod)   {}
+func (f blockingAdditionHandler) HandlePodReconcile(pods []*v1.Pod) {}
+func (f blockingAdditionHandler) HandlePodSyncs(pods []*v1.Pod)     {}
+func (f blockingAdditionHandler) HandlePodCleanups() error          { return nil }
+
+func TestWatchdog(t *testing.T) {
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	defer testKubelet.Cleanup()
+	kubelet := testKubelet.kubelet
+	kubelet.runtimeState.setRuntimeSync(time.Now())
+
+	// setup a fake systemd notify server
+	testDir, e := ioutil.TempDir("/tmp/", "test-")
+	if e != nil {
+		panic(e)
+	}
+	defer os.RemoveAll(testDir)
+	notifySocket := testDir + "/notify-socket.sock"
+	laddr := net.UnixAddr{
+		Name: notifySocket,
+		Net:  "unixgram",
+	}
+	listener, err := net.ListenUnixgram("unixgram", &laddr)
+	require.NoError(t, err, "Failed to setup fake systemd server")
+
+	// configure watchdog
+	wusec := 3000000 // 3 secs
+	os.Setenv("NOTIFY_SOCKET", notifySocket)
```

Comment:
can you also add `defer os.Unsetenv`'s to these too please - otherwise adding other tests that interact with this code will potentially behave unexpectedly.